File: Selection_Sort/Selection_Sort.py
# selection sorting
def find_smallest(arr):
  smallest = arr[0]
  smallest_index = 0
  for i in range(1, len(arr)):
    if arr[i] < smallest:
      smallest = arr[i]
      smallest_index = i
  return smallest_index;
  
  
my_list = [1, 6, -5, 45, -56, 3, 1, -8, -89, -6, -3, 15]

print(find_smallest(my_list))

def selection_sort(arr):
  newArr = []
  for i in range(len(arr)): 
    smallest = find_smallest(arr)
    newArr.append(arr.pop(smallest))
  return newArr
  
print(selection_sort(my_list))

File: Selection_Sort/selectionSort.js
myArr = [1, 3, -5, 6, 6, 14, 15, 689, 13, 11, -26, -356, -1.03, 56, 49, 97, 78];


function selectionSort(arr) {
  for(let i = 0; i < arr.length - 1; i++) {
    let minIndex = i;
    
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }
    
    [ arr[i], arr[minIndex] ] = [ arr[minIndex], arr[i] ];
  }
  
  return arr;
}

console.log(selectionSort(myArr))